src/controllers/product.controller.js

```diff
@@ -1,40 +1,33 @@
 import { collectionServices } from '../services';
 
 const CreateCollection = async (req, res) => {
-  try {
-    const collectionName = req.body.name;
-    const userId = req.user.id;
-    const collectionObj = {
-      userId,
-      name: collectionName,
-    };
+  const collectionName = req.body.name;
+  const userId = req.user.id;
+  const collectionObj = {
+    userId,
+    name: collectionName,
+  };
+  const oldCollection = await collectionServices.getCollectionByName(
+    collectionObj.name
+  );
+  if (!oldCollection) {
     const collection = await collectionServices.createCollection(collectionObj);
     return res.status(201).json({
       code: 201,
       message: 'Collection Created.',
       collection,
     });
-  } catch (error) {
-    return res.status(500).json({
-      code: 500,
-      message: 'Internal Error.',
-      error,
-    });
   }
 };
 
 const DeleteCollection = async (req, res) => {
-  try {
-    const deletedCollection = await collectionServices.deleteCollection(
-      req.params.cid
-    );
-    if (deletedCollection) {
-      return res.status(200).json({ code: 200, message: 'Collection Deleted' });
-    }
-  } catch (error) {
+  const { cid } = req.params;
+  const collection = await collectionServices.getCollectionById(cid);
+  if (collection) {
+    const deletedCollection = await collectionServices.deleteCollection(cid);
     return res
-      .status(500)
-      .json({ code: 500, message: 'Collection Not Deleted', error });
+      .status(200)
+      .json({ code: 200, message: 'Collection Deleted', deletedCollection });
```

Comment:
don't remove try and catch. what if there is an error? the server would crash you need trycatch to handle errors.

Reply:
okay sure